setup/pubsub/google.md

```diff
@@ -0,0 +1,201 @@
+---
+layout: single
+title:  "Google Cloud Pub/Sub"
+sidebar:
+  nav: setup
+---
+
+{% include toc %}
+
+Spinnaker pipelines can be triggered using a [Google Pub/Sub
+subscription](https://cloud.google.com/pubsub/docs/overview).
+
+# Prerequisites
+
+You need a [Google Cloud Platform](https://cloud.google.com/) (GCP) project to
+with the [Pub/Sub API
+enabled](https://pantheon.corp.google.com/apis/api/pubsub.googleapis.com), and
+[`gcloud`](https://cloud.google.com/sdk/downloads) installed locally.  You can
+check that `gcloud` is installed and authenticated by running:
+
+```bash
+gcloud info
+```
+
+## A Pub/Sub Subscription
+
+The Pub/Sub integretion can be used to either:
+
+1. Start pipelines by sending Pub/Sub messages from a system you own.
+
+2. Start pipelines using Pub/Sub messages originating from a Google Cloud
+   system, such as [Google Cloud Storage](https://cloud.google.com/storage/),
+   or [Google Container
+   Registry](https://cloud.google.com/container-registry/).
+
+Configuration and setup for all cases are described below.
+
+### Sending your own Pub/Sub messages
+
+First, record the fact that your `$MESSAGE_FORMAT` is `CUSTOM`, this will be
+needed later.
+
+```bash
+MESSAGE_FORMAT=CUSTOM
+```
+
+You need a topic with name `$TOPIC` to publish messages to:
+
+```bash
+gcloud beta pubsub topics create $TOPIC
+```
+
+This topic needs a a pull subscription named `$SUBSCRIPTION` to let Spinnaker
+read messages from. _It is important that Spinnaker is the only system reading
+from this single subscription. You can always create more subscriptions for
+this topic if you want multiple systems to recieve the same messages._
+
+```bash
+gcloud beta pubsub subscriptions create $SUBSCRIPTION --topic $TOPIC
+```
+
+At this point, you can use the [publisher
+guide](https://cloud.google.com/pubsub/docs/publisher) to learn how to publish
+messages programmatically to the topic you have created.
+
+### Receiving messages from Google Cloud Storage (GCS)
+
+First, record the fact that your `$MESSAGE_FORMAT` is `GCS`, this will be
+needed later.
+
+```bash
+MESSAGE_FORMAT=GCS
+```
+
+Given that you'll be listening to changes in a GCS bucket (`$BUCKET`), the
+following command will create (or use an existing) topic with name `$TOPIC` to
+publish messages to:
+
+```bash
+gsutil notification create -t $TOPIC -f json gs://${BUCKET}
+```
+
+Finally, create a pull subscription named `$SUBSCRIPTION` to listen to changes
+to this topic:
+
+```bash
+gcloud beta pubsub subscriptions create $SUBSCRIPTION --topic $TOPIC
+```
+
+To understand the format that GCS will publish messages in, please read the
+[reference
+material](https://cloud.google.com/storage/docs/pubsub-notifications).
+
+### Receiving messages from Google Container Registry (GCR)
+
+First, record the fact that your `$MESSAGE_FORMAT` is `GCR`, this will be
+needed later.
+
+```bash
+MESSAGE_FORMAT=GCR
+```
```

Comment:
Would it make sense to add this command to the top of the scripts where they're used? For clarity?

Reply:
I'm not sure how to format it, since each of the different systems take slightly different parameters, and the differences don't neatly fit in a table.